File: test_kis.py
import requests
import zipfile
import io

urls = [
    "https://new.real.download.dws.korinvest.com/common/master/kospi_code.mst.zip",
    "https://new.real.download.dws.korinvest.com/common/master/kosdaq_code.mst.zip"
]

for url in urls:
    try:
        print(f"Testing {url}...")
        r = requests.get(url, headers={"User-Agent": "Mozilla/5.0"}, timeout=10)
        print(f"Status: {r.status_code}, Length: {len(r.content)}")
        if r.status_code == 200:
            with zipfile.ZipFile(io.BytesIO(r.content)) as z:
                print(f"Files: {z.namelist()}")
                first_file = [f for f in z.namelist() if f.endswith('.mst')][0]
                with z.open(first_file) as f:
                    content = f.read(100).decode('cp949', errors='ignore')
                    print(f"Sample: {content}")
    except Exception as e:
        print(f"Error: {e}")

File: test_kis.ts
import { JSZip } from "https://deno.land/x/jszip@0.11.0/mod.ts";

const urls = [
  "https://new.real.download.dws.korinvest.com/common/master/kospi_code.mst.zip",
  "https://new.real.download.dws.korinvest.com/common/master/kosdaq_code.mst.zip"
];

const decoder = new TextDecoder("euc-kr");

async function test() {
  for (const url of urls) {
    try {
      console.log(`Testing ${url}...`);
      const response = await fetch(url, {
        headers: { "User-Agent": "Mozilla/5.0" }
      });
      if (!response.ok) throw new Error(`HTTP ${response.status}`);
      
      const arrayBuffer = await response.arrayBuffer();
      console.log(`Success! Length: ${arrayBuffer.byteLength}`);
      
      const zip = new JSZip();
      const unzipped = await zip.loadAsync(arrayBuffer);
      const fileName = Object.keys(unzipped.files).find(f => f.endsWith('.mst'));
      console.log(`File in zip: ${fileName}`);
      
      const contentBuffer = await unzipped.files[fileName].async("uint8array");
      const content = decoder.decode(contentBuffer);
      console.log(`Content sample (first 100 chars): ${content.substring(0, 100)}`);
    } catch (err) {
      console.error(`Failed ${url}: ${err.message}`);
    }
  }
}

test();
